Modules/selectmodule.c

```diff
@@ -433,7 +433,7 @@ select_select_impl(PyObject *module, PyObject *rlist, PyObject *wlist,
 
 typedef struct {
     PyObject_HEAD
-    PyObject *dict;
+    PyObject *dict;     // cannot create cycles as it only contains exact ints
```

Comment:
Am I correct to assume this?

Reply:
Is this change related to making the types immutable?

Reply:
This is correct.